magento/pwa-studio
# [Bugfix] PageBuilder buttons and links load full page

## Description

Fixes a bug in the `resolveLinkProps()` utility where it would silently fail all the time and always returns an object with the `href` property.

Adds a click handler to PageBuilder components that dangerously set HTML content to intercept link clicks and determine whether it is internal or external to avoid unnecessary page reloads.

## Related Issue

Closes PWA-2142

## Acceptance

Any core developers

### Verification Stakeholders

Any core developers

### Specification




## Verification Steps

1. Make sure changes in [this PR](https://github.com/magento-commerce/venia-sample-data-modules/pull/9) has been deployed to the backend instance.
2. Start the dev server: `yarn watch:venia`
3. Navigate to the home page
4. Verify home page renders
5. Click on any of the button/links on the home page
6. Verify page did not do a full reload (either by monitoring the console or network tab)

#### Test scenario(s) for direct fix/feature

1. Verify internal links do not reload the entire page
2. Verify external links still work (this may require you to make changes in the backend)

#### Test scenario(s) for any existing impacted features/areas

1. Verify front page still looks correct
2. Verify buttons/links still work

#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
